### PR TITLE
Add support for Razer Ornata V3 X with PID 02A2

### PR DESCRIPTION
**Pull request by phoerselmann**
This PR adds support for the keyboard Razer Ornata V3 X with the PID 02A2 and should fix this issue #2058

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,7 +96,8 @@ The devices below are fully feature supported by OpenRazer, which means all avai
 | Razer Blade 14 (2022)                         |  1532:028C  |
 | Razer DeathStalker V2 Pro (Wireless)          |  1532:0290  |
 | Razer DeathStalker V2 Pro (Wired)             |  1532:0292  |
-| Razer Ornata V3 X                             |  1532:0294  |
+| Razer Ornata V3 X (0294)                      |  1532:0294  |
+| Razer Ornata V3 X (02A2)                      |  1532:02A2  |
```
Comment (z3ntu) [inline README.md]: Keep the device names here as they are on the pacakging, so I guess just "Razer Ornata V3 X". Also keep the list sorted

File: driver/razerkbd_driver.c [hunk 2]
```diff
@@ -982,8 +983,12 @@ static ssize_t razer_attr_read_device_type(struct device *dev, struct device_att
...
 
-    case USB_DEVICE_ID_RAZER_ORNATA_V3_X:
-        device_type = "Razer Ornata V3 X\n";
+    case USB_DEVICE_ID_RAZER_ORNATA_V3_X_0294:
+        device_type = "Razer Ornata V3 X 0294\n";
+        break;
+
+    case USB_DEVICE_ID_RAZER_ORNATA_V3_X_02A2:
+        device_type = "Razer Ornata V3 X 02A2\n";
         break;
```
Comment (z3ntu) [inline driver/razerkbd_driver.c:989]: Follow something like https://github.com/openrazer/openrazer/blob/master/driver/razerkraken_driver.c#L205-L208

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 3]
```diff
@@ -794,6 +794,24 @@ class RazerOrnataV3X(_MacroKeyboard):
...
+    USB_VID = 0x1532
+    USB_PID = 0x02A2
+    HAS_MATRIX = True
+    MATRIX_DIMS = [1, 1]
+    METHODS = ['get_device_type_keyboard', 'set_static_effect', 'set_spectrum_effect',
+               'set_none_effect', 'set_breath_single_effect', 'set_breath_dual_effect',
+               'set_custom_effect', 'set_key_row', 'get_game_mode', 'set_game_mode', 'get_macro_mode', 'set_macro_mode',
+               'get_macro_effect', 'set_macro_effect', 'get_macros', 'delete_macro', 'add_macro']
+
+    DEVICE_IMAGE = "https://dl.razerzone.com/src/6071-1-en-v1.png"
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/keyboards.py]: make up something like https://github.com/openrazer/openrazer/blob/master/daemon/openrazer_daemon/hardware/headsets.py#L41 if the devices need the same data

File: driver/razerkbd_driver.c [hunk 4]
```diff
@@ -3742,7 +3756,19 @@ static int razer_kbd_probe(struct hid_device *hdev, const struct hid_device_id *
...
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_effect_breath);          // Breathing effect
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_effect_static);          // Static effect
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_effect_custom);          // Custom effect
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_matrix_custom_frame);           // Set LED matrix
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_game_led_state);                // Enable game mode & LED
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_macro_led_state);               // Enable macro LED
+            CREATE_DEVICE_FILE(&hdev->dev, &dev_attr_macro_led_effect);              // Change macro LED effect (static, flashing)
+            break;
+
+        case USB_DEVICE_ID_RAZER_ORNATA_V3_X_02A2:
```
Comment (z3ntu) [inline driver/razerkbd_driver.c]: also here, if the case statements are the same, combine them together

File: driver/razerkbd_driver.h [hunk 5]
```diff
@@ -93,7 +93,8 @@
 #define USB_DEVICE_ID_RAZER_BLADE_14_2022 0x028C
 #define USB_DEVICE_ID_RAZER_DEATHSTALKER_V2_PRO_WIRELESS 0x0290
 #define USB_DEVICE_ID_RAZER_DEATHSTALKER_V2_PRO_WIRED 0x0292
-#define USB_DEVICE_ID_RAZER_ORNATA_V3_X 0x0294
+#define USB_DEVICE_ID_RAZER_ORNATA_V3_X_0294 0x0294
+#define USB_DEVICE_ID_RAZER_ORNATA_V3_X_02A2 0x02A2
```
Comment (z3ntu) [inline driver/razerkbd_driver.h]: keep the list sorted

File: README.md (hunk 1 above)
Comment (phoerselmann) [reply README.md]: Has been changed

File: driver/razerkbd_driver.c (hunk 2 above)
Comment (phoerselmann) [reply driver/razerkbd_driver.c:989]: Thank you for the link. modified like for the kraken device.

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 3 above)
Comment (phoerselmann) [reply daemon/openrazer_daemon/hardware/keyboards.py]: changed like the proposed solution

File: driver/razerkbd_driver.c (hunk 4 above)
Comment (phoerselmann) [reply driver/razerkbd_driver.c]: also , changed like the proposed solution

File: driver/razerkbd_driver.h (hunk 5 above)
Comment (phoerselmann) [reply driver/razerkbd_driver.h]: done

**Comment by z3ntu** [comment]
Looks good now. Just to confirm, all the functionality shown in a UI like polychromatic works properly? No issues seen?

**Comment by phoerselmann** [comment]
Yes, all functions are working good. I tested my changes the last days and everything went well.

**Review by z3ntu — APPROVED** [review]
Great! Thanks for the patch then, LGTM!

**Comment by ShinkarenkoMaxim** [comment]
Everything works great! Thank you! The only thing Polychromatic does not detect is brightness when pressing keys when the controller is open.

**Comment by z3ntu** [comment]
@ShinkarenkoMaxim I don't get what you mean. Could you rephrase?

**Comment by VlaK0r** [comment]
> Looks good now. Just to confirm, all the functionality shown in a UI like polychromatic works properly? No issues seen?

Please tell me how to apply this fix?

**Comment by z3ntu** [comment]
@VlaK0r Either wait for the next release, or install openrazer from nightly repo (Ubuntu daily PPA / openrazer-git in AUR) or https://github.com/openrazer/openrazer/wiki/Building-a-package

**Comment by VlaK0r** [comment]
> @VlaK0r Either wait for the next release, or install openrazer from nightly repo (Ubuntu daily PPA / openrazer-git in AUR) or https://github.com/openrazer/openrazer/wiki/Building-a-package

Thank you. And when is the next release?

**Comment by z3ntu** [comment]
There's no fixed schedule but since the last release has been a while ago, maybe in the coming 1 or 2 months.

**Comment by VlaK0r** [comment]
> @VlaK0r Either wait for the next release, or install openrazer from nightly repo (Ubuntu daily PPA / openrazer-git in AUR) or https://github.com/openrazer/openrazer/wiki/Building-a-package

Please tell me how to install nightly repo on debian?

**Comment by z3ntu** [comment]
There is none, but you should be able to follow the Debian instructions on that page and just remove the `-b TARGET_BRANCH` part from the instructions.
